### PR TITLE
AP_Mount: split servo driver, parameters to degrees

### PR DESCRIPTION
**Pull request by rmackay9**
This includes the following changes to the AP_Mount library

- Parameters are moved to a separate AP_Mount_Params class similar to what we've done for RangeFinders and Proximity sensors.  Structurally this is nice and it also makes increasing the number of mount instances very easy,  This leads to MNT_xxx parameters being renamed to MNT1_ although we could leave the first instance as MNT_ because there are currently no class-level parameters.
- Parameters for angle limits are renamed and scaled in degrees (instead of centi-degrees).  The default angle limits have also been updated to match most users setups
    - MNT_ANGMIN_ROL -> MNT1_ROLL_MIN   (default was -4500, now -30 to match Copter's ANGLE_MAX)
    - MNT_ANGMAX_ROL -> MNT1_ROLL_MAX   (default was +4500, now +30)
    - MNT_ANGMIN_TIL -> MNT1_PITCH_MIN   (default was -4500, now -90)
    - MNT_ANGMAX_TIL -> MNT1_PITCH_MAX   (default was -4500, now +20)
    - MNT_ANGMIN_PAN -> MNT1_YAW_MIN   (default was -4500, now -180)
    - MNT_ANGMAX_PAN -> MNT1_YAW_MAX   (default was +4500, now +180)
- RC channels used to control of the mount(s) (in RC_TARGETTING mode) are specified using RCx_OPTION parameters instead of MNT_RC_IN_ROLL,  MNT_RC_IN_TILT and MNT_RC_IN_PAN
- A new "BrushlessPWM" mount type is added that uses the existing Servo backend.  This allows users to more simply specify that their gimbal does not need roll and pitch stabilization instead of setting the MNT_STAB_ROLL and MNT_STAB_TILT parameters.  The MNT_STAB_PAN parameter has also been removed along with some complex rotation code because it is no longer necessary since earlier PRs have cleaned up the earth-frame vs body-frame confusion and given users the ability to switch between these two frames in real time (see "Mount Lock" RC option)

This has been lightly tested in SITL on a Gremsy gimbal

To-Do:

- [x] Add parameter conversion
- [x] Test SToRM32, AlexMos
- [x] Test Solo gimbal

### CODE REVIEW

**Comment by rmackay9** [comment]
I've now added in the parameter conversion and tested it thoroughly

Below are screen shots of the three parameter conversions tests that were done.

The top image shows the changes from centi-degrees to degrees and also how MNT_TYPE may change from 1/Servo to 7/BrushlessPWM.  The 2nd image shows how MNT_TYPE may stay as 1.
mnt-params-before-vs-after-annotated

The next image shows the successful conversion from the RC controls from MNT_RC_IN_PAN/ROLL/TILT to RCx_OPTION
mnt-rc-conversion-before-vs-after

File: libraries/AP_Mount/AP_Mount.cpp [hunk 1]
```diff
@@ -879,11 +501,88 @@ void AP_Mount::handle_gimbal_device_attitude_status(const mavlink_message_t &msg
...
+    }
+
+    // find Mount's top level key
+    uint16_t k_param_mount_key;
+    if (!AP_Param::find_top_level_key_by_pointer(this, k_param_mount_key)) {
+        return;
+    }
+
+    // table of mount parameters to be converted without scaling
+    const AP_Param::ConversionInfo mnt_param_conversion_info[] = {
```
Comment (peterbarker) [inline libraries/AP_Mount/AP_Mount.cpp:539]:
```suggestion
    static const AP_Param::ConversionInfo mnt_param_conversion_info[] {
```

File: libraries/AP_Mount/AP_Mount.cpp [hunk 2]
```diff
@@ -879,11 +501,88 @@ void AP_Mount::handle_gimbal_device_attitude_status(const mavlink_message_t &msg
...
+        { k_param_mount_key, 17, AP_PARAM_FLOAT, "MNT1_LEAD_RLL" },
+        { k_param_mount_key, 18, AP_PARAM_FLOAT, "MNT1_LEAD_PTCH" },
+    };
+    uint8_t table_size = ARRAY_SIZE(mnt_param_conversion_info);
+    for (uint8_t i=0; i<table_size; i++) {
+        AP_Param::convert_old_parameter(&mnt_param_conversion_info[i], 1.0f);
+    }
+
+    // mount parameters conversion from centi-degrees to degrees
+    const AP_Param::ConversionInfo mnt_param_deg_conversion_info[] = {
```
Comment (peterbarker) [inline libraries/AP_Mount/AP_Mount.cpp:552]:
```suggestion
    static const AP_Param::ConversionInfo mnt_param_deg_conversion_info[] {
```

File: libraries/AP_Mount/AP_Mount_Servo.h [hunk 3]
```diff
@@ -55,6 +56,9 @@ class AP_Mount_Servo : public AP_Mount_Backend
     ///  moves servo with the given function id to the specified angle.  all angles are in body-frame and degrees * 10
     void move_servo(uint8_t rc, int16_t angle, int16_t angle_min, int16_t angle_max);
 
+    /// Servo gimbals require stabilization, BrushlessPWM gimbals self-stabilize
+    bool requires_stabilization;
```
Comment (tridge) [inline libraries/AP_Mount/AP_Mount_Servo.h:60]: const

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Mount/AP_Mount.cpp (hunk 1 above)
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount.cpp:539]: fixed, txs.

File: libraries/AP_Mount/AP_Mount.cpp (hunk 2 above)
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount.cpp:552]: fixed, txs.

File: libraries/AP_Mount/AP_Mount_Servo.h (hunk 3 above)
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Servo.h:60]: ok txs.  made "requires_stabilization" const.

**Pull request merged**